Source/Scene/ImageryLayer.js

```diff
@@ -969,6 +1149,230 @@ define([
         }
     };
 
+    function MakeCommandOptions() {
+        this.context = undefined;
+        this.outputTexture = undefined;
+        this.imagery = undefined;
+        this.imageryLayer = undefined;
+        this.vertexArray = undefined;
+        this.projectedRectangle = undefined;
+        this.projectedTexture = undefined;
+        this.final = undefined;
+    }
+
+    var arbitraryReprojectAttributeIndices = {
+        position : 0,
+        projectedCoordinates : 1
+    };
+
+    var geographicCartographicScratch = new Cartographic();
+    var projectedScratch = new Cartesian3();
+
+    /**
+     * Enqueues commands re-projecting multiple textures to a {@link GeographicProjection} on the next update
+     * and generate mipmaps for the geographic texture.
+     */
+    ImageryLayer.prototype._multisourceReprojectTexture = function(frameState, imagery) {
+        var projectedTextures = imagery.projectedTextures;
+        var projectedRectangles = imagery.projectedRectangles;
+        var projectedTexturesLength = projectedTextures.length;
+        var someTexture = projectedTextures[0];
+
+        var width = someTexture.width;
+        var height = someTexture.height;
+
+        var outputTexture = imagery.texture = new Texture({
+            context : frameState.context,
+            width : width,
+            height : height,
+            pixelFormat : someTexture.pixelFormat,
+            pixelDatatype : someTexture.pixelDatatype,
+            preMultiplyAlpha : someTexture.preMultiplyAlpha
+        });
+
+        if (CesiumMath.isPowerOfTwo(width) && CesiumMath.isPowerOfTwo(height)) {
+            outputTexture.generateMipmap(MipmapHint.NICEST);
```

Comment:
I don't think this does anything. Plus, the mipmaps will be generated in `finalizeReprojectTexture`.

Reply:
Ah, I see the other comment about generating mipmaps. Could you add that here as well?